Source/DafnyCore/AST/Expressions/Variables/NonglobalVariable.cs

```diff
@@ -50,6 +50,9 @@ public abstract class NonglobalVariable : TokenNode, IVariable {
   /// <returns>A string uniquely determined from <c>nm</c>, containing none of the characters in
   /// <c>specialChars</c>.</returns>
   public static string SanitizeName(string nm) {
+    if (nm.Length == 0) {
```

Comment:
What's the use of this??

Reply:
If `nm` is empty, the 0/9 check below would result in array index out of bounds exception.
It seems reasonable that the sanitized version of `""` is still `""`.

I encountered this during development when my `python-module-name` started with a `.`.
I don't have this any more, but I think the fix still applies.
I'm happy to remove if this is wrong though.

Reply:
I'd remove it or replace it with an explicit exception, since the implicit precondition of the method is that the name is non-empty (which makes sense since an empty symbol is pretty much never valid). Otherwise you might hide/obfuscate future similar bugs.

Reply:
Removed